cli/src/program.rs

```diff
@@ -2749,6 +2789,30 @@ fn report_ephemeral_mnemonic(words: usize, mnemonic: bip39::Mnemonic) {
     eprintln!("[BUFFER_ACCOUNT_ADDRESS] argument to `solana program close`.\n{divider}");
 }
 
+fn set_compute_budget_ixs_if_needed(ixs: &mut Vec<Instruction>, compute_unit_price: &Option<u64>) {
+    let mut compute_units_required_for_ixs: u32 = 0;
+    for ix in ixs.iter() {
+        if ix.program_id == bpf_loader_upgradeable::id() {
+            compute_units_required_for_ixs += 2370u32;
+        } else if ix.program_id == system_program::id() {
+            compute_units_required_for_ixs += 150u32;
+        } else {
+            compute_units_required_for_ixs += 2000u32;
+        }
+    }
+
+    let compute_units_for_compute_budget_ixs = 300u32;
+
+    if let Some(compute_unit_price) = compute_unit_price {
+        ixs.push(ComputeBudgetInstruction::set_compute_unit_limit(
+            compute_units_required_for_ixs + compute_units_for_compute_budget_ixs,
+        ));
+        ixs.push(ComputeBudgetInstruction::set_compute_unit_price(
+            *compute_unit_price,
+        ));
```

Comment:
compute budget instructions need to be inserted at the beginning of the instruction list